backend/app/api/knowledge_bases.py

```
@@ -700,10 +1190,40 @@ def promote_knowledge_base_to_overall(
 @router.post("/{knowledge_base_id}/rollback")
 def rollback_knowledge_base(
     knowledge_base_id: str,
-    request: KnowledgeBaseRollbackRequest,
+    request: KnowledgeBaseRollbackRequest | SharedKnowledgeRollbackRequest,
     db: Session = Depends(get_session),
```

Ruff (0.16.2)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)